### ISSUE TITLE
Remove old TEM functionality from FV dynamical core

### ISSUE THREAD

**Issue by fvitt** [ESCOMP/CAM#774]
### Issue Type

Code Clean-up

### Issue Description

Remove the deprecated TEM diagnostics scheme from the finite volume dycore code.  This is superseded by the TEM physics grid scheme.   The `*zm` history field names will be adopted by the new physics TEM scheme.

### Will this change answers?

No

### Will you be implementing this yourself?

Yes

**Comment by adamrher** [comment]
Could @dan800 or @islasimpson make a statement of verification that the FV zonal mean diagnostics that we are deprecating here, can be more-or-less reproduced using the new zonal mean method? Have we ran two FV runs, one with new, one with old? I just want to be certain before we make this irreversible decision

**Comment by islasimpson** [comment]
I agree it would be worth running old and new in an FV run and checking them.  I don't know if that has been done already as I haven't really been involved in this.  I expect @dan800 knows.

**Comment by fvitt** [comment]
I will hold off on this for now.  However, I think the new TEM diags scheme can use the `*zm` history fields names.  We just cannot use both the old and new TEM schemes in the same run.

We did the testing @islasimpson is suggesting when developing the new scheme.

**Comment by adamrher** [comment]
> We did the testing @islasimpson is suggesting when developing the new scheme.

That's good enough for me. Thanks.
